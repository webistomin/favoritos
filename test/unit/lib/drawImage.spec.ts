import Favoritos from '../../../src/favoritos';

describe('Favoritos: drawImage method', () => {
  beforeEach(() => {
    document.head.innerHTML = `<link rel="shortcut icon" href="#">`;
  });

  it('Draw image to canvas if content is correct', () => {
    const lib = new Favoritos();
    const correctContent = new Image();
    expect(() => lib['drawImage'](correctContent)).not.toThrow(TypeError);
    // @ts-ignore
    expect(lib['iconCanvasContext'].__getEvents()).toMatchSnapshot();
  });

  it('Ignore setting "crossorigin" anonymous attribute on canvas', () => {
    new Favoritos();
    const correctContent = document.createElement('canvas');
    // @ts-ignore
    expect(correctContent.crossOrigin).toEqual(undefined);
  });

  it('Throw error if content is not "CanvasImageSource" type', () => {
    const lib = new Favoritos();
    const correctContent = 'something wrong';
    expect(() => lib['drawImage']((correctContent as unknown) as CanvasImageSource)).toThrow(TypeError);
    // @ts-ignore
    expect(lib['iconCanvasContext'].__getEvents()).toMatchSnapshot();
  });
});
