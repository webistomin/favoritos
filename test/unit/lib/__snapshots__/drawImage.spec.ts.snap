// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favoritos: drawImage method draw image to canvas if content is correct 1`] = `
Array [
  Object {
    "props": Object {
      "value": "18px Helvetica, Arial, sans-serif",
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "font",
  },
  Object {
    "props": Object {
      "value": "center",
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "textAlign",
  },
  Object {
    "props": Object {
      "value": "middle",
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "textBaseline",
  },
  Object {
    "props": Object {
      "x": 1,
      "y": 1,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "scale",
  },
  Object {
    "props": Object {
      "dHeight": 0,
      "dWidth": 0,
      "dx": 0,
      "dy": 0,
      "img": <img
        crossorigin="anonymous"
      />,
      "sHeight": 0,
      "sWidth": 0,
      "sx": 0,
      "sy": 0,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "drawImage",
  },
]
`;

exports[`Favoritos: drawImage method throw error if content is not CanvasImageSource type 1`] = `
Array [
  Object {
    "props": Object {
      "value": "18px Helvetica, Arial, sans-serif",
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "font",
  },
  Object {
    "props": Object {
      "value": "center",
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "textAlign",
  },
  Object {
    "props": Object {
      "value": "middle",
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "textBaseline",
  },
  Object {
    "props": Object {
      "x": 1,
      "y": 1,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "scale",
  },
]
`;
